packages/cozy-intent/src/api/services/NativeService.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { WebviewRef } from 'api/models/environments'
+import { NativeEvent } from 'api/models/events'
+import { NativeMethodsRegister } from 'api/models/methods'
+import { NativeMessenger } from '../services/NativeMessenger'
+import { NativeService } from './NativeService'
+
+jest.mock('../services/NativeMessenger')
+
+const onMessageMock = jest.fn()
+class MockNativeMessenger extends NativeMessenger {
+  constructor(webviewRef: WebviewRef) {
+    super(webviewRef)
+  }
+
+  public postMessage = jest.fn()
+
+  public addMessageListener = jest.fn()
+
+  public onMessage = (event: NativeEvent): void => {
+    onMessageMock(event)
+  }
+}
+
+describe('NativeMessenger', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('Should allow to register and unregister webviews', async () => {
+    const nativeMethods: NativeMethodsRegister = {
+      logout: jest.fn(),
+      openApp: jest.fn()
+    }
+
+    const webviewRef: WebviewRef = {
+      injectJavaScript: jest.fn(),
+      props: {
+        source: {
+          uri: 'http://SOME_URI'
+        }
+      }
+    }
+
+    const nativeService = new NativeService(nativeMethods, MockNativeMessenger)
+
+    nativeService.registerWebview(webviewRef)
+
+    await nativeService.tryEmit({
+      nativeEvent: { data: '{"source": "post-me"}', url: 'http://SOME_URI' }
+    })
+
+    expect(onMessageMock).toHaveBeenCalledTimes(1)
```

Comment:
expect(onMessageMock).toHaveBeenNthCalledWith(1, 'something')

Reply:
Thanks, replaced in https://github.com/cozy/cozy-libs/compare/18c6fb9c2fc90ac667c8e10b42f29552f2dd360b..c8724f304b52d9a21b6cb7a9aea1ce0165e236e9